### PR TITLE
[PLAY-145]Caption Kit Convert Typescript

### PR DESCRIPTION
**Pull request by ericawinne**
React Kit Coverage
#### Breaking Changes: 
no

#### Runway Ticket URL:  
https://nitro.powerhrg.com/runway/backlog_items/PLAY-145

#### How to test this: 
Check kit in Milano Environment 

#### Checklist:

- [X] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [X] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [X] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

**Review by ericawinne — COMMENTED** [review]
color, id, and text do not need = ''. Makes sense for color and id, why not text?

**Comment by nidaqg** [comment]
Look awesome Erica!

File: playbook/app/pb_kits/playbook/pb_caption/_caption.tsx [hunk 1]
```diff
@@ -1,24 +1,22 @@
-/* @flow */
-
 import React from 'react'
 import classnames from 'classnames'
 import { buildAriaProps, buildCss, buildDataProps } from '../utilities/props'
-import { deprecatedProps, globalProps } from '../utilities/globalProps'
+import { deprecatedProps, globalProps, GlobalProps } from '../utilities/globalProps'
```
Comment (RudeChowder) [inline playbook/app/pb_kits/playbook/pb_caption/_caption.tsx:4]: We have a `globalProps` and a `GlobalProps`? That feels a bit confusing

**Review by RudeChowder — APPROVED** [review]
Nice work!

File: playbook/app/pb_kits/playbook/pb_caption/_caption.tsx (hunk 1 above)
Comment (ericawinne) [reply playbook/app/pb_kits/playbook/pb_caption/_caption.tsx:4]: It's a bit confusing for me as well. I know it saved me from having to specify a bunch of props 'GlobalProbs' already has.

Comment (ericawinne) [reply playbook/app/pb_kits/playbook/pb_caption/_caption.tsx:4]: I'll get clarification and report back 🙃

Comment (thestephenmarshall) [reply playbook/app/pb_kits/playbook/pb_caption/_caption.tsx:4]: @RudeChowder nah `GlobalProps` is the type

File: playbook/app/pb_kits/playbook/pb_caption/_caption.tsx [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
   color?: "default" | "light" | "lighter" | "success" | "error" | "link",
-  data?: object,
+  data?: {[key: string]: string},
   id?: string,
   size?: "xs" | "sm" | "md" | "lg" | "xl",
   tag?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "p" | "span" | "div" | "caption",
   text?: string,
   variant?: null | "link",
-};
+} & GlobalProps;
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_caption/_caption.tsx:17]: @RudeChowder and you can see it in action here

**Review by thestephenmarshall — APPROVED** [review]

File: playbook/app/pb_kits/playbook/pb_caption/_caption.tsx (hunk 1 above)
Comment (ericawinne) [reply playbook/app/pb_kits/playbook/pb_caption/_caption.tsx:4]: oppps.. I think I misunderstood what's actually going on here.. 👀 maybe we can talk about this one.  thanks @thestephenmarshall

File: playbook/app/pb_kits/playbook/pb_caption/_caption.tsx (hunk 2 above)
Comment (RudeChowder) [reply playbook/app/pb_kits/playbook/pb_caption/_caption.tsx:17]: ah, I see. So `GlobalProps` is the type definitions and `globalProps` is the actual props

**Review by diontheroulde — APPROVED** [review]

**Comment by diontheroulde** [comment]
LGTM!

**Pull request merged**